### PR TITLE
Android: Only show divider in settings for headers

### PR DESCRIPTION
**Pull request by K0bin**
Before:
image

After:
image

I think it looks much cleaner. I can't request a review, so I'll just ping @t895.

### CODE REVIEW

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/settings/ui/SettingsDividerItemDecoration.kt [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+  override fun onDraw(
+    canvas: Canvas, parent: RecyclerView, state: RecyclerView.State
+  ) {
+    if (parent.layoutManager == null) {
+      return
+    }
+
+    canvas.save()
+    var left: Int
+    var right: Int
```
Comment (t895) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/settings/ui/SettingsDividerItemDecoration.kt:77]:
```suggestion
    val left: Int
    val right: Int
```
These are only assigned once

**Comment by t895** [comment]
Could you apply this to the cheats list too?

**Comment by t895** [comment]
Could you also apply the built-in codestyle? You're using two spaces for indentation when it should be 4.

**Review by t895 — APPROVED** [review]
LGTM

**Review by mbc07 — APPROVED** [review]

**Comment by MayImilae** [comment]
I don't feel like this is necessarily the perfect solution, as there is some clarity issues with it like this, but the dividers as they are are way too strong and busy. Either the dividers need a total rework or they need to be removed. So, I think this is a net win even if it isn't "perfect".

LGTM.

**Pull request merged**